src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -5849,6 +5837,19 @@ SyntaxKind.FileScopedNamespaceDeclaration or
 
             AddInitializers(builder.InstanceInitializers, instanceInitializers);
             AddInitializers(builder.StaticInitializers, staticInitializers);
+
+            void reportMisplacedMemberInNamespace(SyntaxNode member, SyntaxToken locationSyntax, bool reportMisplacedGlobalCode)
+            {
+                if (IsImplicitClass && reportMisplacedGlobalCode)
+                {
+                    var parentSyntax = member.Parent;
+                    var errorCode = parentSyntax.IsKind(SyntaxKind.CompilationUnit)
+                        && parentSyntax.ChildNodes().Any(node => node.IsKind(SyntaxKind.GlobalStatement))
```

Comment:
this will do a linear walk on the compilation unit each time.   instead of this, i would pass in the parent as the preceding argument (before 'members') in AddNonTypeMembers.  Then compute and store if the comp-unit has global statements.  (can just do `parent is CompilationUnit compilationUnit && compilationUnit.Members.Any(m => m is GlobalStatementSyntax)`).